src/coreclr/jit/compiler.cpp

```diff
@@ -4080,6 +4080,15 @@ const char* Compiler::compGetTieringName(bool wantShortName) const
 {
     const bool tier0 = opts.jitFlags->IsSet(JitFlags::JIT_FLAG_TIER0);
     const bool tier1 = opts.jitFlags->IsSet(JitFlags::JIT_FLAG_TIER1);
+
+    if (!opts.compMinOptsIsSet)
+    {
+        // If 'compMinOptsIsSet' is not set, just return here. Otherwise, if this method is called
+        // by the assertAbort(), we would recursively call assert while trying to get MinOpts()
+        // and eventually stackoverflow.
+        return "Optimization-Level-Not-Yet-Set";
+    }
```

Comment:
So now it's possible we previously might have returned "Tier0" or "Tier1" but now we'll return this instead. Is that ok?

Reply:
IMO - I think it should be fine given the rare scenario in which it occurs. This will only happen if we hit an assert before completing `compInit()`. The main scenario for which it was originally added would still return the expected value.